Source/Particles/Collision/MCCProcess.H

```diff
@@ -84,33 +49,85 @@ public:
     static
     void readCrossSectionFile (
                                const std::string cross_section_file,
-                               VectorType<amrex::Real>& energies,
-                               VectorType<amrex::Real>& sigmas
+                               amrex::Vector<amrex::Real>& energies,
+                               amrex::Gpu::HostVector<amrex::Real>& sigmas
                                );
 
     static
     void sanityCheckEnergyGrid (
-                                const VectorType<amrex::Real>& energies,
+                                const amrex::Vector<amrex::Real>& energies,
                                 amrex::Real dE
                                 );
 
-    MCCProcessType m_type;
-    amrex::Real m_energy_penalty;
+    struct Executor {
+        /** Get the collision cross-section using a simple linear interpolator. If
+         * the energy value is lower (higher) than the given energy range the
+         * first (last) cross-section value is used.
+         *
+         * @param E_coll collision energy in eV
+         *
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        amrex::Real getCrossSection (amrex::Real E_coll) const
+        {
+            if (E_coll < m_energy_lo) {
+                return m_sigma_lo;
+            } else if (E_coll > m_energy_hi) {
+                return m_sigma_hi;
+            } else {
+                using amrex::Math::floor;
+                using amrex::Math::ceil;
+                // calculate index of bounding energy pairs
+                amrex::Real temp = (E_coll - m_energy_lo) / m_dE;
+                int idx_1 = static_cast<int>(floor(temp));
+                int idx_2 = static_cast<int>(ceil(temp));
+
+                // linearly interpolate to the given energy value
+                temp -= idx_1;
+                return m_sigmas_data[idx_1] + (m_sigmas_data[idx_2] - m_sigmas_data[idx_1]) * temp;
+            }
+        }
+
+        amrex::Real* m_sigmas_data;
```

Comment:
Just a safety (to crash reproducibility) since we manually init the data later on:
```suggestion
        amrex::Real* m_sigmas_data = nullptr;
```